inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/ClassInstrumentationConfiguration.java

```diff
@@ -0,0 +1,58 @@
+package rocks.inspectit.oce.core.instrumentation.config;
+
+import lombok.Getter;
+import net.bytebuddy.description.type.TypeDescription;
+import rocks.inspectit.oce.core.instrumentation.special.SpecialSensor;
+
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Defines the actual or requested state of the instrumentation of a single class.
+ */
+public class ClassInstrumentationConfiguration {
+
+    /**
+     * The configuration representing that no instrumentation of the class is performed.
+     */
+    public static final ClassInstrumentationConfiguration NO_INSTRUMENTATION = new ClassInstrumentationConfiguration(Collections.emptySet(), null);
+
+    /**
+     * The sensors which are active for the target class.
+     */
+    @Getter
+    private final Set<SpecialSensor> activeSpecialSensors;
+
+    /**
+     * The inspectIT configuration used to derive this configuration object.
+     * Can be null, but only if no sensors active.
+     */
+    @Getter
+    private final InstrumentationConfiguration activeConfiguration;
+
+    public ClassInstrumentationConfiguration(Set<SpecialSensor> activeSpecialSensors, InstrumentationConfiguration activeConfiguration) {
+        this.activeSpecialSensors = activeSpecialSensors;
+        this.activeConfiguration = activeConfiguration;
+    }
+
+    /**
+     * Compares this instrumentation configuration against another.
+     * Two instrumentations are considered to be the "same" if they result in the same bytecode changes.
+     *
+     * @param forType the type for which this configuration is meant.
+     * @param other   the other configuration to compare against
+     * @return true if both are the same, false otherwise
+     */
+    public boolean isSameAs(TypeDescription forType, ClassInstrumentationConfiguration other) {
+        if (!activeSpecialSensors.equals(other.activeSpecialSensors)) {
```

Comment:
as you removed the static method now these collections could be different, go for the `Collections.disjoint()` or make constructor package private?

Reply:
Disjoint btw would be semantically incorrect! Disjoint only returns true, if they have not a single element in common. I added a simple implementation to CommonUtils